pages/settings/attributes/[attributeId]/edit.tsx

```diff
@@ -0,0 +1,61 @@
+import {
+  DialogContent,
+  DialogTitle,
+} from '@mui/material'
+import AttributeForm, { AttributeFormData } from 'components/AttributeForm'
+import { useRouter } from 'next/router'
+import React from 'react'
+import { useEffect, useState } from 'react'
+import { AttributeResponse } from 'utils/types'
+
+export default function AttributeEditForm() {
+  // you have to do this to otherwise the AttributeForm says that
+  // attribute is being used before its given a value
+  const [attribute, setAttribute] = useState<AttributeResponse>()
+  const router = useRouter()
+  const { id } = router.query
+  useEffect(() => {
+    const fetchAttribute = async () => {
+      if (!id) return // on page load, id is undefined, resulting in bad requests
+      fetch(`/api/attributes/${id}`, { method: 'GET' }).then((response) => {
+        response.json().then((data) => {
+          setAttribute(data.payload)
+        })
+      })
+    }
+    fetchAttribute()
+  }, [id])
+
+  const handleSubmit = async (
```

Comment:
Problem with `handleSubmit` or the `onChange` definition in `UpsertAttributeForm`... getting a type error when trying to edit
![image](https://user-images.githubusercontent.com/41850520/235573441-236a1533-699e-4a63-a52a-ef3018102032.png)